### DISCUSSION TITLE
await?

### DISCUSSION THREAD

**Discussion by thomhurst** [dotnet/csharplang#8627] [Language Ideas] [answered]
Sometimes we want to await an object if it's not null, but awaiting a null object will give a NRE.

So we either:
- Surround the Task in a not null check if statement
- use the null coalescing operator to fallback to a Completed task

Both options are just more verbose/boilerplate than necessary.

Example:

```
await (stringWriter?.DisposeAsync() ?? ValueTask.CompletedTask);
```

Or

```
if (stringWriter is not null)
{
    await stringWriter.DisposeAsync();
}
```

These could be simplified to just:

```
await? stringWriter?.DisposeAsync();
```

The single question mark is already known as the null forgiving operator, so it would be clear what it's trying to achieve. 

The compiler can easily rewrite it to just perform the if statement for us, so it's syntactic sugar.

**Comment by timcassell** [comment, marked as answer]
#35
